### ISSUE TITLE
Helpers object includes limited methods

### ISSUE THREAD

**Issue by 2collegebums** [markevans/block_helpers#1]
in a block helper class, the helper object contains only the barebones helper methods and is missing critical functions such as the ability to call 'capture' within the block helper. In addition, it is missing all custom helper methods even when they exist within the application.

Seems like your own helpers should be included such that you can use the helper object the same way you would work with a standard helper method.

Ideally, there could be a delegation so that you don't need to call helpers.content_tag and instead can just call 'content_tag' directly with the method being delegated automatically.

UPDATE:
the best option is to do what a FormBuilder does with @template. This is done by passing in the ActionView::Base instance itself. We think the best way to do this is to create an additional helper method which automatically passes in the self instance and stores it within the BlockHelper instance.

example:

```
def widget_item(&block)
    widget_item_helper(self, &block)
end

class WidgetItemHelper < ActionView::BlockHelper
    def initialize(template)
      @template = template
    end
    ....
end
```

rendered with: 
    widget_item do |w| ...

**Comment by markevans** [comment]
Hi thanks for your comments

Funnily enough I just pushed a change to have capture and friends available directly instead of via the 'helper' method (although I've kept it there for use in case of name clashes).

I originally had it like that, but I changed it to the protected 'helper' object because I didn't like them being public (it's difficult to include privately/protected, because some helper methods may not be defined at the time that the class is declared, which is when all the stuff happens, using self.inherited)
However I don't really care that they're public now - I don't think it matters.

I don't know what you mean about missing 'capture' etc. though - that should be available. Which version of rails are you using it on?

Also, I like the way you suggested about doing what form builder does, only I don't want to have to create a separate helper method and class, it's nice to have the helper method created automatically using the camelcase => underscore naming convention, so I've included it directly.
Let me know if you know how to do it like form builder without defining a separate helper method and class
